File: src/data/courses/dsa/binary-search-on-answer/rate-speed-problems/minimum-time-complete-trips/code/python/minimum_time_complete_trips.py
class Solution:
    def minimumTime(self, time, totalTrips):
        lo, hi = 1, max(time) * totalTrips
        while lo < hi:
            mid = (lo + hi) // 2
            trips = sum(mid // t for t in time)
            if trips >= totalTrips: hi = mid
            else: lo = mid + 1
        return lo

File: src/data/courses/dsa/binary-search-on-answer/rate-speed-problems/minimum-time-complete-trips/code/java/MinimumTimeCompleteTrips.java
class Solution {
    public long minimumTime(int[] time, int totalTrips) {
        long lo=1,hi=0; for(int t:time)hi=Math.max(hi,t); hi*=totalTrips;
        while(lo<hi){long mid=lo+(hi-lo)/2,trips=0;for(int t:time)trips+=mid/t;if(trips>=totalTrips)hi=mid;else lo=mid+1;}
        return lo;
    }
}